### PR TITLE
Fix ping command not supporting api.php as the value for pingDefaultTestPage

### PR DESCRIPTION
**Pull request by s-k-z**
For users who ran Kolmafia between #1821 and #1828, `pingDefaultTestPage` was set to `api.php` for their default value and likely never changed. For those users, the ping command doesn't work for two reasons:

* PingRequest's generic request URL gets set to `(none)` when it fails to match a string ending in .php
* `isSafeToRun()` returns false for any `pingURL` that ends in .php

Example failure:
```
> get pingDefaultTestPage

api.php

> ping

Ping returned no response; ping test aborted
0 pings to api.php at 0-0 msec apiece (total = 0, average = 0.00) = 0 bytes/second
```
After this change:
```
> ping

10 pings to api.php at 79-95 msec apiece (total = 846, average = 84.60) = 23,191 bytes/second

> ping

10 pings to api at 81-103 msec apiece (total = 910, average = 91.00) = 21,560 bytes/second
```

This PR addresses both issues and makes the command operable again under the above condition. Additionally, I eliminated some warnings in the `PingRequest.java` file.

### CODE REVIEW

**Comment by Veracity0** [comment]
I'm all for backwards compatibility, but "api.php" is ambiguous.
Is it "status" or "events'?
I going to have to consider your proposed solution.

**Comment by s-k-z** [comment]
Ah, yeah this one just loads the API help page. Since it resolves as a real page maybe that's fine?

File: src/net/sourceforge/kolmafia/request/PingRequest.java [hunk 1]
```diff
@@ -12,18 +12,15 @@ public class PingRequest extends GenericRequest {
   // api.php's responseText is about 1/4 the size of main.php's - and
   // measured ping time is about 1/4 as long.
 
-  public PingRequest() {
```
Comment (Veracity0) [inline src/net/sourceforge/kolmafia/request/PingRequest.java:15]: Why did you delete this constructor?

File: src/net/sourceforge/kolmafia/request/PingRequest.java [hunk 2]
```diff
@@ -117,14 +115,6 @@ protected boolean shouldFollowRedirect() {
     return false;
   }
 
-  public long getStartTime() {
```
Comment (Veracity0) [inline src/net/sourceforge/kolmafia/request/PingRequest.java:120]: Again, why did you delete these "getters"?

**Review by Veracity0 — CHANGES_REQUESTED** [review]

**Comment by Rinn** [comment]
if it's just to fix those users why don't we just update the property if 'api.php' is set and reset back to the default 'api' then in like a month delete the code that resets the property.

File: src/net/sourceforge/kolmafia/request/PingRequest.java (hunk 2 above)
Comment (s-k-z) [reply src/net/sourceforge/kolmafia/request/PingRequest.java:120]: It's unused as far as I can tell.

File: src/net/sourceforge/kolmafia/request/PingRequest.java (hunk 1 above)
Comment (s-k-z) [reply src/net/sourceforge/kolmafia/request/PingRequest.java:15]: This constructor isn't used as far as I can tell.

**Comment by Veracity0** [comment]
> if it's just to fix those users why don't we just update the property if 'api.php' is set and reset back to the default 'api' then in like a month delete the code that resets the property.

I've been away from KoL and my development computer for 2.5 weeks.

I approve of this suggestion. Migrate the setting and delete (or not) the code to do that after a suitable period.

**Comment by Rinn** [comment]
I've just added a temp hack to reset the property
